**Урок 5. Применение математической статистики для проверки гипотез в реальной жизни для популярных метрик**

1.

Для оценки статистической значимости различий между конверсиями в двух группах при α=5% можно использовать Z-критерий для разности долей. Формула Z-критерия:
Z = (p1 - p2) / sqrt(p * (1 - p) * (1/N1 + 1/N2))
где p = (n1 * p1 + n2 * p2) / (n1 + n2)


In [3]:
from math import sqrt

p1 = 164 / 15550
p2 = 228 / 15550
n1 = 164
n2 = 228
N1 = 15550
N2 = N1
p =  (n1 * p1 + n2 * p2) / (n1 + n2)
Z = (p1 - p2) / sqrt(p * (1 - p) * (1/N1 + 1/N2))

-3.2110930329358367

*Ответ:*

Критическое значение Z-критерия при α=5% и двухсторонней альтернативе равно 1.96. Так как расчетное значение Z меньше критического значения, т.е. -3.2110930329358367 < 1.96 , то мы отвергаем нулевую гипотезу о равенстве конверсий в группах. То есть, изменения внесенные в заказную форму в магазине Утконос дали статистически значимый эффект на уровень конверсии в покупку. Рекомендуется использовать новую форму заказа.

2.Посчитайте контрольное число: среднее из 4 чисел (выборочную дисперсию, 
выборочное среднее, стандартное отклонение и медиану) для выборки (первые 10 натуральных разных чисел). 


In [1]:
import statistics


In [2]:
data = [3, 7, 1, 4]


### выборочную дисперсию:

In [3]:
statistics.variance(data)

6.25

### выборочное среднее:

In [5]:
statistics.mean(data)

3.75

### стандартное отклонение:

In [6]:
statistics.pstdev(data)

2.165063509461097

### медиана:

In [7]:
statistics.median(data)

3.5